apps/ledger-live-mobile/src/screens/BleDevicePairingFlow/BleDeviceScanning.tsx

```diff
@@ -0,0 +1,172 @@
+    filterByDeviceModelIds: filterByDeviceModelId
+      ? [filterByDeviceModelId]
+      : undefined,
+    filterOutDevicesByDeviceIds: areKnownDevicesDisplayed ? [] : knownDeviceIds,
+  });
+
+  // Handles scanning error
+  useEffect(() => {
+    if (scanningBleError) {
+      // Currently using the error code values from react-native-ble-plx
+      // It should be defined indenpendently, in live-common
+      if (
+        scanningBleError?.errorCode === BleErrorCode.LocationServicesDisabled
+      ) {
+        setStopBleScanning(true);
+        setLocationDisabledError(true);
+      }
+
+      if (scanningBleError?.errorCode === BleErrorCode.BluetoothUnauthorized) {
+        setStopBleScanning(true);
+        setLocationUnauthorizedError(true);
+      }
+    }
+  }, [scanningBleError]);
+
+  const renderItem = useCallback(
+    ({ item }: { item: ScannedDevice }) => {
+      const isAlreadyKnown = !!(
+        areKnownDevicesDisplayed &&
+        knownDeviceIds.some(deviceId => deviceId === item.deviceId)
+      );
+
+      const deviceMeta = {
+        deviceId: item.deviceId,
+        deviceName: `${item.deviceName}`,
+        wired: false,
+        modelId: item.deviceModel.id,
+        isAlreadyKnown,
+      };
+
+      return (
+        <BleDeviceItem
+          deviceMeta={deviceMeta}
+          onSelect={() => onDeviceSelect(item)}
+        />
+      );
+    },
+    [areKnownDevicesDisplayed, knownDeviceIds, onDeviceSelect],
+  );
+
+  const onLocationFixed = useCallback(() => {
+    setLocationDisabledError(false);
+    setLocationUnauthorizedError(false);
+    setStopBleScanning(false);
+  }, [setLocationDisabledError, setLocationUnauthorizedError]);
+
+  if (locationDisabledError) {
+    return <LocationRequired onRetry={onLocationFixed} errorType="disabled" />;
+  }
+
+  if (locationUnauthorizedError) {
+    return (
+      <LocationRequired onRetry={onLocationFixed} errorType="unauthorized" />
+    );
+  }
+
+  return (
+    <Flex bg="background.main" height="100%">
+      <Flex px={4}>
+        <Flex mb={8} alignItems="center">
+          <BluetoothThingy />
+        </Flex>
+        <Text mb={3} textAlign="center" variant="h4" fontWeight="semiBold">
+          {productName
+            ? t("blePairingFlow.scanning.withProductName.title", {
+                productName,
+              })
+            : t("blePairingFlow.scanning.withoutProductName.title")}
+        </Text>
+        <Text
+          mb={8}
+          color="neutral.c70"
+          textAlign="center"
+          variant="body"
+          fontWeight="medium"
+        >
+          {productName
+            ? t("blePairingFlow.scanning.withProductName.description", {
+                productName,
+              })
+            : t("blePairingFlow.scanning.withoutProductName.description")}
+        </Text>
+        <FlatList
+          data={scannedDevices}
+          renderItem={renderItem}
+          keyExtractor={item => `${item.deviceId}-${Math.random()}`}
```

Comment:
why the Math.random? isn't deviceId enough?

Reply:
as long as the `deviceId` are unique, yes i agree

Reply:
are they not?

Reply:
They are, except if someone uses a custom BLE device with a duplicated id - but that would be rare. And would only cause a display error in the listing. So i put it at `item.deviceId` only ;)